5.5/root/usr/share/container-scripts/mysql/validate-variables.sh

```diff
@@ -32,11 +30,34 @@ function validate_variables() {
     root_specified=1
   fi
 
+  # If MYSQL_USER == "root", we have a special case
+  if [[ "${user_specified:-0}" == "1" && "$MYSQL_USER" == "root" ]]; then
+    if [[ "${root_specified:-0}" == "1" ]]; then
+      usage "You cannot set MYSQL_USER to 'root' and use MYSQL_PASSWORD and MYSQL_ROOT_PASSWORD at the same time"
```

Comment:
nm, I see what you're saying/doing here... if mysql_user==root, then mysql_password and mysql_root_password become equivalent, so specifying both is an error.

reword the message to:
usage "When setting MYSQL_USER to 'root' you can only set either MYSQL_PASSWORD or MYSQL_ROOT_PASSWORD"

though I still think all this behavior is confusing and would have gone for just reporting an error if they set MYSQL_USER to root and say "don't do that, to configure the root user just use MYSQL_ROOT_PASSWORD and do not specify MYSQL_USERNAME"